tests/test_integration.py

```diff
@@ -88,6 +88,31 @@ def circuit(x, y, z):
 
         assert np.allclose(circuit(a, b, c), expected, **tol)
 
+    def test_gradient_for_tensor_product(self):
+        """Test that the gradient of a circuit containing a tensor product is
+        computed without any errors."""
+        n_qubits = 2
+        depth = 2
+
+        def ansatz(weights):
+            weights = weights.reshape(depth, n_qubits, 3)
+            qml.templates.StronglyEntanglingLayers(weights, wires=list(range(n_qubits)))
+            return qml.expval(qml.PauliZ(0) @ qml.PauliZ(1))
+
+        dev_qsk = qml.device(
+                    "qiskit.aer",
+                    wires=n_qubits,
+                    shots=1000,
+                    backend="qasm_simulator",
+                )
+
+        weights = np.random.random((depth, n_qubits, 3)).flatten()
+
+        # Want to get expectation value and gradient
+        exp_sampled = qml.QNode(ansatz, dev_qsk, diff_method="parameter-shift")
+        grad_shift = qml.grad(exp_sampled, argnum=0)
+        exp_sampled(weights)
+        grad_shift(weights)
```

Comment:
Awesome!

Reply:
(However this test is missing asserts)

Reply:
Thanks! Yes this one is just meant to check that no errors arise, as this does not work now with the `master` version

Reply:
Ah, right! I didn't see the docstring properly, and thought this test was unfinished :)